src/Symfony/Component/Console/Style/SymfonyStyle.php

```diff
@@ -320,4 +363,47 @@ private function getTerminalWidth()
 
         return $dimensions[0] ?: self::MAX_LINE_LENGTH;
     }
+
+    private function autoPrependBlock()
+    {
+        $chars = substr($this->bufferedOutput->fetch(), -2);
+
+        if (false === $chars) {
+            return $this->newLine(); //empty history, so we should start with a new line.
+        }
+        //Prepend new line for each non PHP_EOL chars (This means no blank line was output before)
+        $this->newLine(2 - substr_count($chars, PHP_EOL));
+    }
+
+    private function autoPrependText()
+    {
+        $chars = $this->bufferedOutput->fetch();
+        //Prepend new line if last char isn't EOL:
+        if (PHP_EOL !== substr($chars, -1)) {
+            $this->newLine();
+        }
+    }
+
+    private function reduceBuffer($messages)
+    {
+        //Retrieve last inserted chars from the two latest inserted lines:
+        $fetched = $this->bufferedOutput->fetch();
+
+        if (is_string($messages)) {
+            if (false === strpos($messages, PHP_EOL)) {
+                //No EOL, should return previous + single message
+                return substr($fetched.$messages, -2);
+            }
+            //Has EOL. Split into multiples messages.
+            $messages = array_map(function ($message) {
+                return $message.PHP_EOL; //Re-add trailing EOL
```

Comment:
I'm wondering if an EOL $messages string can exist, which would mean an empty string here (before the concat)